test/unit/finance/no_variable_cost_test.rb

```diff
@@ -1,16 +1,51 @@
 require 'test_helper'
 
 class Finance::NoVariableCostClass < ActiveSupport::TestCase
+  attr_reader :contract, :account, :app_plan
 
-  def test_bill_for_variable
-    contract = FactoryBot.build_stubbed(:contract)
-    account  = FactoryBot.build_stubbed(:simple_account)
-    app_plan = FactoryBot.build_stubbed(:application_plan)
+  setup do
+    @contract = FactoryBot.build_stubbed(:contract)
+    @account  = FactoryBot.build_stubbed(:simple_account)
+    @app_plan = FactoryBot.build_stubbed(:application_plan)
+
+    @org_tz = ENV["TZ"]
+  end
+
+  teardown do
+    ENV["TZ"] = @org_tz
+  end
+
+  test "bill for variable" do
+    Timecop.travel(15.days.ago) if Time.now.mday == 1
```

Comment:
I do not get that :)
Why not set a fixed date instead?

Reply:
I can use fixed time. But we caught the issue because test failed at some certain time. So I wanted to let it run at unexpected times and timezones so that we have a chance to catch unexpected behavior. Only at the times we know it should fail, alter time to something else. Let me know.